assembl/static2/js/app/components/debate/thread/answerForm.jsx

```diff
@@ -94,9 +95,22 @@ class AnswerForm extends React.PureComponent<*, AnswerFormProps, AnswerFormState
         };
         displayAlert('success', I18n.t('loading.wait'));
         createPost({ variables: variables })
-          .then(() => {
-            refetchIdea().then(() => {
-              this.setState({ body: null }, hideAnswerForm);
+          .then((res) => {
+            const postId = res.data.createPost.post.id;
+            this.setState({ submitting: false, body: null }, () => {
+              hideAnswerForm();
+              // Execute refetchIdea after the setState otherwise we can get a warning setSate called on unmounted component.
+              refetchIdea().then(() => {
+                // The thread may have moved because it became the latest active, we need to scroll to the created post.
+                // Arbitrary wait 200ms to be sure the post is rendered.
```

Comment:
I understand this 200ms might be sufficient for our above average devices/internet. But, going based on this feels hacky to me. Is it possible to have an event listener at the parent component of the message list, and on a post's `componentDidMount` trigger a `postRendered` method, which any other component can observe upon and do actions (like this one)? 

Downside, more event management, likely perf hit, boilerplate code. Upside: assured scroll-to's, readable code??

Thoughts?

Reply:
Here's a thought:

https://stackoverflow.com/a/28748160

Reply:
Ah requestAnimationFrame may do the trick if we're sure that the box is painted on next frame, thanks. I'll test that.